alf/algorithms/agent.py

```diff
@@ -356,11 +379,15 @@ def after_train_iter(self, experience, train_info: AgentInfo = None):
         """
         algorithms = [
             self._rl_algorithm, self._representation_learner,
-            self._goal_generator
+            self._goal_generator, self._reward_weight_algorithm
         ]
         algorithms = list(filter(lambda a: a is not None, algorithms))
         self._agent_helper.after_train_iter(algorithms, experience, train_info)
 
+        if self._reward_weight_algorithm:
```

Comment:
Can document the expected behavior for the case where a ``reward_weight_algorithm`` is used and a ``reward_weights`` is also specified for the ``rl_algorithm``.

Reply:
added documentation